src/devices/Mcp3xxx/Mcp3xxx.cs

```diff
@@ -0,0 +1,158 @@
+        /// </summary>
+        /// <remarks>
+        /// Like a normal differential input the value that is read respresents the difference between the voltage on the value channel and the voltage on the reference channel (valueChannel Reading - referenceChannel Reading).
+        /// However the reference signal in a pseudo-differential input is expected to be connected to the signal ground. This is used to reduce the effect of external electrical noise on the on the inputs. If there is noise where
+        /// the noise is likey to impact both the value input and the reference input and the action of subtracting the values helps to cancel it out.
+        /// </remarks>
+        /// <param name="valueChannel">Channel which represents the signal (valid values: 0 to channelcount - 1).</param>
+        /// <param name="referenceChannel">Channel which represents the signal ground (valid values: 0 to channelcount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public virtual int ReadPseudoDifferential(int valueChannel, int referenceChannel)
+        {
+            // ensure that the channels are part of the same pairing
+            CheckChannelPairing(valueChannel, referenceChannel);
+
+            // read and return the value. the value passsed to the channel represents the channel pair. 
+            return ReadInternal(channel: valueChannel / 2, valueChannel > referenceChannel ? InputType.InvertedDifferential : InputType.Differential, _adcResolutionBits);
+        }
+
+        /// <summary>
+        /// Reads a value from the device using differential inputs
+        /// </summary>
+        /// <remarks>
+        /// The value that is read respresents the difference between the voltage on the value channel and the voltage on the reference channel (valueChannel Reading - referenceChannel Reading).
+        /// This subtraction is performed in software which may mean that errors are introduced with rapidly changing signals.
+        /// </remarks>
+        /// <param name="valueChannel">Channel which represents the signal driving the value in a positive direction (valid values: 0 to channelcount - 1).</param>
+        /// <param name="referenceChannel">Channel which represents the signal driving the value in a negative direction (valid values: 0 to channelcount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public virtual int ReadDifferential(int valueChannel, int referenceChannel)
+        {
+            CheckChannelRange(valueChannel, ChannelCount);
+            CheckChannelRange(referenceChannel, ChannelCount);
+
+            if (valueChannel == referenceChannel)
+            {
+                throw new ArgumentException($"ADC differential channels must be different.", nameof(valueChannel) + " " + nameof(referenceChannel));
+            }
```

Comment:
should there also be equivalent of:

```csharp
if (ChannelsSupportPseudoDifferential(valueChannel, referenceChannel))
{
    return ReadPseudoDifferential(valueChannel, referenceChannel);
}
```

Reply:
I don't believe so as you wouldn't want to do a pseudo-differential read in place of a real one. With pseudo-differential on these chipsets the referenceValue is limited to +/- 100mv and so you probably wouldn't get the same answer.